File: android/src/main/java/com/video/hybrids/HybridVideoPlayerSourceFactory.kt
package com.margelo.nitro.video

import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class HybridVideoPlayerSourceFactory: HybridVideoPlayerSourceFactorySpec() {
  override fun fromUri(uri: String): HybridVideoPlayerSourceSpec {
    return HybridVideoPlayerSource(uri)
  }

  override val memorySize: Long
    get() = 0
}

File: ios/hybrids/HybridVideoPlayerSourceFactory.swift
//
//  HybridVideoPlayerSourceFactory.swift
//  NitroVideo
//
//  Created by Krzysztof Moch on 23/09/2024.
//

import Foundation

class HybridVideoPlayerSourceFactory: HybridVideoPlayerSourceFactorySpec {
  func fromUri(uri: String) -> HybridVideoPlayerSourceSpec {
    return HybridVideoPlayerSource(uri: uri)
  }
  
  
  // Initialize HybridContext
  var hybridContext = margelo.nitro.HybridContext()
  
  // Return size of the instance to inform JS GC about memory pressure
  var memorySize: Int {
    return getSizeOf(self)
  }
}
